test/suite/test_hs21.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import time, re
+import wiredtiger, wttest
+from wtdataset import SimpleDataSet
+from wiredtiger import stat
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_hs21.py
+# Test we don't lose any data when idle files with an active history are closed/sweeped.
+# Files with active history, ie content newer than the oldest timestamp can be closed when idle.
+# We want to ensure that when an active history file is idle closed we can continue reading the
+# correct version of data and their base write generation hasn't changed (since we haven't
+# restarted the system).
+class test_hs21(wttest.WiredTigerTestCase):
+    # Configure handle sweeping to occur within a specific amount of time.
+    conn_config = 'file_manager=(close_handle_minimum=0,close_idle_time=3,close_scan_interval=1),' + \
+            'statistics=(all),operation_tracking=(enabled=false)'
+    session_config = 'isolation=snapshot'
+    file_name = 'test_hs21'
+    numfiles = 10
+    nrows = 10000
+
+    def large_updates(self, uri, value, ds, nrows, commit_ts):
+        # Update a large number of records, we'll hang if the history store table isn't working.
+        session = self.session
+        cursor = session.open_cursor(uri)
+        for i in range(1, nrows + 1):
+            session.begin_transaction()
```

Comment:
You can instead write all values in the same transaction:

```
begin_transaction()
for i in range(..)
    cursor[..] = value
commit_transaction()
```

Reply:
Yep true, I've updated it to be in a single transaction